packages/core/contracts-ovm/insured-bridge/implementation/L1_BridgeRouter.sol

```diff
@@ -0,0 +1,127 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity >=0.7.6;
+
+import "@eth-optimism/contracts/libraries/bridge/OVM_CrossDomainEnabled.sol";
+
+import "./L2_BridgeDepositBox.sol";
+
+/**
+ * @notice Contract deployed on L1 that has an implicit reference to a DepositBox on L2 and provides methods for
+ * "Relayers" to fulfill deposit orders to that contract. The Relayers can either post capital to fulfill the deposit
+ * instantly, or request that the funds are taken out of the passive liquidity provider pool following a challenge period.
+ * @dev A "Deposit" is an order to send capital from L2 to L1, and a "Relay" is a fulfillment attempt of that order.
+ */
+contract BridgeRouter is OVM_CrossDomainEnabled {
+    // Finder used to point to latest OptimisticOracle and other DVM contracts.
+    address public finder;
+
+    // Links deposit contract addresses to L2 networks they are deployed on.
+    // TODO: Can we convert uint256 to uint128 or smaller based on some assumption about how large netIDs get?
+    mapping(uint256 => address) public netIdToDepositContract;
+
+    // Links L2-L1 addresses between canonical versions of the same token for each network. For example, if the
+    // official address of WETH on L2 is 0x123 and the official address of WETH on L1 is 0xabc, then the mapping will
+    // be (0x123 => 0xabc).
+    mapping(address => address) public whitelistedTokens;
+
+    // Set upon construction and can be reset by Owner.
+    uint256 public optimisticOracleLiveness;
+
+    // A Deposit represents a transfer that originated on an L2 DepositBox contract and can be bridged via this contract.
+    enum DepositState { PendingSlow, PendingInstant, FinalizedSlow, FinalizedInstant }
+    enum DepositType { Slow, Instant }
+
+    struct Deposit {
+        DepositState depositState;
+        DepositType depositType;
+        // The following params are set by the L2 depositor:
+        address l1Recipient;
+        address l2Token;
+        uint256 amount;
+        uint256 maxFee;
+        // Params inferred by this contract:
+        address l1Token;
+        // The following params are inferred and set by the L2 deposit contract:
+        address l2Sender;
+        uint256 depositTimestamp;
+        uint256 depositNetId;
+        // Relayer will compute the realized fee considering the amount of liquidity in this contract and the pending
+        // withdrawals at the depositTimestamp.
+        uint256 realizedFee;
+        // A deposit can have both a slow and an instant relayer if a slow relay is "sped up" from slow to instant. In
+        // these cases, we want to store both addresses for separate payouts.
+        address slowRelayer;
```

Comment:
I renamed "initialRelayer" to "slowRelayer" because its more consistent with the `DepositState` and `DepositType` but let me know if the `initialRelayer` is NOT always the slow relayer.